Encourage to overwrite CommitScheduler.push_to_hub (#1506)
huggingface/huggingface_hub: docs/source/guides/upload.mdx

    `scheduler.lock` lock to ensure thread-safety. The lock is blocked only when the scheduler scans the folder for
    changes, not when it uploads data. You can safely assume that it will not affect the user experience on your Space.

### Space persistence demo

Persisting data from a Space to a Dataset on the Hub is the main use case for [`CommitScheduler`]. Depending on the use
case, you might want to structure your data differently. The structure has to be robust to concurrent users and
restarts which often implies generating uuids. Beside robustness, you would ideally like to upload data in a format that
is readable by the `datasets` library for later reuse. We compiled in this [interactive demo Space](https://huggingface.co/spaces/Wauplin/space_to_dataset_saver)
implementations to tackle major use cases. Depending on your specific needs, you might need to adapt them

### Custom uploads

As said above, [`CommitScheduler`] assumes that your data is append-only and should be uploading "as is". However, you

> might want to customize the way data is uploaded. You can do that by creating a class inheriting from [`CommitScheduler`]
> and overwrite the `push_to_hub` method. You can overwrite this method the way you want. You are guaranteed that it will

Reviewer: ```suggestion
might want to customize the way data is uploaded. You can do that by creating a class inheriting from [`CommitScheduler`]
and overwrite the `push_to_hub` method (feel free to overwrite it any way you want). You are guaranteed it will
```

be called every `every` minutes in a background thread. You don't have to worry about concurrency and errors but you

> must be careful other aspects like not pushing empty commits or duplicated data.

Reviewer: ```suggestion
must be careful about other aspects, such as pushing empty commits or duplicated data.
```
